src/DocLinkChecker/Program.cs

```diff
@@ -455,16 +455,36 @@ private static void ProcessFile(DirectoryInfo folder, string filepath)
                                                 // Get rid of the title mark
                                                 var lineTitleLink = lineTitle.Replace("#", string.Empty);
 
+                                                // check if there is an HTML link ID like this: ## <a id=\"i_am_an_id\">Title for ID</a>
+                                                Regex idLinkPattern = new Regex(@"(?><a id=\"")(?<id_label>[a-zA-Z -_]+)(?>\"">)");
```

Comment:
the regex should be trying to find out if `something.md#title` is correct.

Looking at the doc, this is the source: `| [D-Decimal](https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-numeric-format-strings#DFormatString) | [Decimal unit test output](string-format-examples.md#d-decimal) |`

And it seems well formed. The regex seems to find something else than that

Reply:
This is not the correct way:
![image](https://user-images.githubusercontent.com/8145578/207330425-1c5bc573-a6bd-4ba4-89c2-431b7855f00e.png)

This is the normal way: `## D-decimal` and that will magically work

Reply:
That regex is working on the target link, not where it's being "called".

Reply:
change all the ID for normal titles and all will be properly working again. All the file called `string-format-examples.md` is not well formed and the tool is right that the link on it is not correct. Remove the ID link, they are useless!

Reply:
According to docfx it's valid: https://dotnet.github.io/docfx/tutorial/links_and_cross_references.html#links-in-inline-html
Unless I'm misinterpreting it...

Reply:
What about content\building\cmake-presets.md 
build-instructions.md#build_version-matching?

This is pointing to content\building\build-instructions.md >>>>  ## BUILD_VERSION matching

And the one with the ? in the title.

Reply:
> build-instructions.md#build_version-matching

![image](https://user-images.githubusercontent.com/8145578/207356854-0e80aa6c-18cc-42c3-8ed8-96e36527395e.png)

as _ are not permitted, should be `build-instructions.md#build-version-matching`

Reply:
note that tools like VS Code are quite permissive with those kind of links and non theoretically supported characters. It will also do its best to show a table even if fully broken.